Apsitvarkom.Api/Controllers/CleaningEventController.cs

```diff
@@ -73,4 +76,36 @@ public async Task<ActionResult<CleaningEventResponse>> GetById([FromQuery] Objec
 
         return Ok(mappedEvent);
     }
+
+    [ProducesResponseType(StatusCodes.Status201Created)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(List<string>))]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    [HttpPost("Create")]
+    public async Task<ActionResult<CleaningEventResponse>> Create(CleaningEventCreateRequest cleaningEventCreateRequest)
```

Comment:
There is an issue with this. If the polluted location id doesn't exist, we return a StatusCode500, which means there has been an internal server error and there was no such internal issue - we should instead return an error code 404 with an error message `Polluted location with the specified id '{cleaningEventCreateRequest.PollutedLocationId}' was not found.`

This can be probably achieved in two ways:
- Checking what kind of exception we get - I believe the `InnerException` in the try/catch of the `InsertAsync` holds a dictionary with `ConstraintName` key, whose value is FK_CleaningEvents_PollutedLocations_PollutedLocationId. If we ever find out, that this is the exact circumstances of the exception that has happened, we would return a different status code.
- Adding a new method to `IRepository` interface `ParentExistsByPropertyAsync`, which would have one purpose, defined by its name. In instances, that do not have any parent (such as the `PollutedLocation`), we would simply throw `NotSupportedException`.

I personally think that the second approach is a better way to go, but I could be wrong, so perhaps we could check with @BenasB about this.

Reply:
I second the second option, was thinking the same while doing Update

Reply:
All other conversations except this are resolved. Lets wait for @BenasB input about this solution.

Reply:
You can check https://github.com/vu-vibedosa/apsitvarkom-be/pull/132 to see how I approached it, this closely resembles what @6ar8nas  advised in the second point I think

Reply:
I agree with the purpose of this comment, some thoughts:
- Probably a [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) instead of a 404
- Second option is the way to go IMO, just not creating `ParentExistsByPropertyAsync` on `IRepository` but on `ICleaningEventRepository` (like we have with `IPollutedLocationRepostory`), then we will not need to throw `NotSupportedException` for polluted location repository. If we can have something determined at compile time, we should go for it.

Reply:
sure. thought injecting another repo might be not ideal, but we would essentially be doing the same, as is already done, so we could pretty much just reuse it instead. let's roll with that @XxDoMaSxX

Reply:
Well I was reading https://stackoverflow.com/questions/48449192/rest-http-response-code-for-missing-parent-resource and I might be incorrect for the status code, so we can stich to 404

Reply:
interesting. then let's have it a 404.